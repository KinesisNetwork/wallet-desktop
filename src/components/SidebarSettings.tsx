import { WalletView } from '@types'
import * as React from 'react'

export interface Props {
  connectionName: string
  changeWalletView: (view: WalletView) => any
}

export const SidebarSettings: React.SFC<Props> = ({ connectionName, changeWalletView }) => (
  <div className='has-text-centered vertical-spaced' style={{ justifyContent: 'flex-end' }}>
    <label className='label is-small'>Connection: {connectionName}</label>
    <button className='button is-outlined is-fullwidth' onClick={() => changeWalletView(WalletView.settings)}>
      Settings
    </button>
    <p className='is-size-4 has-text-primary' style={{ marginTop: '1rem' }}>Kinesis Wallet - alpha</p>
  </div >
)

SidebarSettings.displayName = 'SidebarSettings'
