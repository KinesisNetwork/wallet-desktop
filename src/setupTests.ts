import { configure } from 'enzyme'
import * as Adapter from 'enzyme-adapter-react-16'

export default configure({ adapter: new Adapter() })
