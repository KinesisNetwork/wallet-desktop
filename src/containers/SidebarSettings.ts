import { connect } from 'react-redux'

import { changeWalletView } from '@actions'
import { SidebarSettings as SidebarSettingsPresentation } from '@components/SidebarSettings'
import { RootState } from '@store'

const mapStateToProps = ({ connections }: RootState) => ({
  connectionName: connections.currentStage as string,
})

export const SidebarSettings = connect(
  mapStateToProps,
  { changeWalletView },
)(SidebarSettingsPresentation)
