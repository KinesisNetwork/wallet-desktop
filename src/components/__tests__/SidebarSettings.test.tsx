import { shallow } from 'enzyme';
import * as React from 'react'

import { SidebarSettings } from '@components/SidebarSettings'
import '../../setupTests'

describe('SidebarSettings', () => {
  let props

  beforeEach(() => {
    props = {
      changeWalletView: () => null,
      connectionName: 'Kinesis KAU Testnet'
    }
  })

  it('renders correctly', () => {
    const wrapper = shallow(<SidebarSettings {...props} />)

    expect(wrapper).toMatchSnapshot()
  })

  it('renders the connection name and a settings button', () => {
    const wrapper = shallow(<SidebarSettings {...props} />)

    const label = wrapper.find('label')
    const button = wrapper.find('button')

    expect(label).toHaveLength(1)
    expect(label.text()).toEqual('Connection: Kinesis KAU Testnet')
    expect(button).toHaveLength(1)
    expect(button.text()).toEqual('Settings')
  })

  it('calls the changeWalletView method when the button is clicked', () => {
    const changeWalletViewMock = jest.fn()
    const wrapper = shallow(<SidebarSettings {...props} changeWalletView={changeWalletViewMock} />)

    const button = wrapper.find('button')
    button.simulate('click')

    expect(changeWalletViewMock).toHaveBeenCalled()
  })
})
