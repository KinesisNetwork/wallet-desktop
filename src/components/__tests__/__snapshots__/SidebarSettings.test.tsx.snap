// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarSettings renders correctly 1`] = `
<div
  className="has-text-centered vertical-spaced"
  style={
    Object {
      "justifyContent": "flex-end",
    }
  }
>
  <label
    className="label is-small is-capitalized"
  >
    Connection: 
    Kinesis KAU Testnet
  </label>
  <button
    className="button is-outlined is-fullwidth"
    onClick={[Function]}
  >
    Settings
  </button>
  <p
    className="is-size-4 has-text-primary"
    style={
      Object {
        "marginTop": "1rem",
      }
    }
  >
    Kinesis Wallet - alpha
  </p>
</div>
`;
